Formula/couchdb.rb

```diff
@@ -60,15 +67,63 @@ def install
     system "make"
     system "make", "install"
 
+    install_geocouch if build.with? "geocouch"
+
     # Use our plist instead to avoid faffing with a new system user.
     (prefix/"Library/LaunchDaemons/org.apache.couchdb.plist").delete
     (lib/"couchdb/bin/couchjs").chmod 0755
-    (var/"lib/couchdb").mkpath
-    (var/"log/couchdb").mkpath
-    (var/"run/couchdb").mkpath
+  end
+
+  def geocouch_share
+    share/"couchdb-geocouch"
+  end
+
+  def install_geocouch
+    resource("geocouch").stage(buildpath/"geocouch")
+    ENV["COUCH_SRC"]="#{buildpath}/src/couchdb"
+    cd "geocouch" do
+      system "make"
+
+      linked_geocouch_share = (HOMEBREW_PREFIX/"share/couchdb-geocouch")
+      geocouch_share.mkpath
+      geocouch_share.install "ebin"
+      #  Install geocouch.plist for launchctl support.
```

Comment:
I don't understand the `plist` and services stuff that well, but this seems like the right thing to do; it seems like when installed `--with-geocouch`, it should always run with the `geocouch` stuff enabled.

I'm not sure what all's involved with making this the actual plist. Should I just move it to `homebrew.mxcl.couchdb.plist` under the prefix? Or slurp it in, process it, and conditionally return it inside the `plist` method? Is `plist()` guaranteed to run in a particular order with respect to `install()`?